custom_platforms.js

```diff
@@ -6,6 +6,459 @@
+ *   it may have a value (`=`), or it may be the last modifier in the rule (`$`).
+ *
+ * Currently it is not supported, but it can be added in the future
+ * https://github.com/AdguardTeam/CoreLibs/issues/1778
+ */
+const PROTOBUF_MODIFIER_PATTERNS = [
+    "\\$(?!((path|domain)=.*])).*protobuf(,|=|$)",
+];
+
+/**
+ * Pattern to check if rule contains `$app` modifier
+ * and does not contain non-basic modifiers like `$domain` or `$path`.
+ *
+ * Pattern parts:
+ * - `\\$` — modifiers divider
+ * - `(?!((path|domain)=.*])).*` — negative lookahead to exclude rules like `[$path=...]##.textad,[app="ads"]`
+ * - `.*app=` — app= modifier itself
+ *
+ * @example
+ * ```@@||imasdk.googleapis.com^$app=tv.htv.app```
+ */
+const APP_MODIFIER_PATTERNS = [
+    "\\$(?!((path|domain)=.*])).*app=",
+];
+
+/**
+ * Pattern to check if rule contains `$extension` modifier
+ * and does not contain non-basic modifiers like `$domain` or `$path`.
+ *
+ * Pattern parts:
+ * - `\\$` — modifiers divider
+ * - `(?!((path|domain)=.*])).*` — negative lookahead to exclude rules like `[$path=...]##.textad,extension`
+ * - `.*extension` — extension modifier itself
+ * - `(,|=|$)` — end of line or modifiers divider, as `$extension` can be followed by other modifiers (`,`),
+ *   it may have a value (`=`), or it may be the last modifier in the rule (`$`).
+ *
+ * @example
+ * ```@@||radar.cloudflare.com^$elemhide,extension,content```
+ */
+const EXTENSION_MODIFIER_PATTERNS = [
+    "\\$(?!((path|domain)=.*])).*extension(,|=|$)",
+];
+
+/**
+ * Pattern to check if rule contains only `$content` modifier.
+ *
+ * @example
+ * ```@@||telegram.hr^$content```
+ */
+const ONLY_CONTENT_MODIFIER_PATTERNS = [
+    "\\$content$",
+];
+
+/**
+ * Pattern to check if rule contains `$content` modifier
+ * and does not contain non-basic modifiers like `$domain` or `$path`.
+ *
+ * Pattern parts:
+ * - `\\$` — modifiers divider
+ * - `(?!((path|domain)=.*])).*` — negative lookahead to exclude rules like `[$path=...]##.textad,content`
+ * - `.*content` — content modifier itself
+ * - `(,|$)` — end of line or modifiers divider, as `$content` can be followed by other modifiers (`,`).
+ *
+ * @example
+ * ```@@||dnsleaktest.com^$content,elemhide,jsinject```
+ */
+const CONTENT_MODIFIER_PATTERNS = [
+    "\\$(?!((path|domain)=.*])).*content(,|$)",
+];
+
+/**
+ * Pattern to check if rule contains `$jsinject` modifier
+ * and does not contain non-basic modifiers like `$domain` or `$path`.
+ *
+ * Pattern parts:
+ * - `\\$` — modifiers divider
+ * - `(?!((path|domain)=.*])).*` — negative lookahead to exclude rules like `[$path=...]##.textad,jsinject`
+ * - `.*jsinject` — jsinject modifier itself
+ * - `(,|$)` — end of line or modifiers divider, as `$jsinject` can be followed by other modifiers (`,`).
+ *
+ * @example
+ * ```@@://www.atlassian.com^$elemhide,jsinject,extension```
+ */
+const JSINJECT_MODIFIER_PATTERNS = [
```

Comment:
may it be added for other extensions, not just ext_ublock?

Reply:
the same for `URLBLOCK_MODIFIER_PATTERNS`

Reply:
Yes, both can be added for `android-content-blocker`.

Reply:
Done.